core/go/internal/publictxmgr/transaction_orchestrator.go

```diff
@@ -220,6 +229,102 @@ func (oc *orchestrator) getFirstInFlight() (ift *inFlightTransactionStageControl
 	return
 }
 
+func (oc *orchestrator) initNextNonceFromDBRetry(ctx context.Context) error {
+	return oc.retry.Do(ctx, func(attempt int) (retryable bool, err error) {
+		return true, oc.initNextNonceFromDB(ctx)
+	})
+}
+
+func (oc *orchestrator) initNextNonceFromDB(ctx context.Context) error {
```

Comment:
On startup of the `orchestrator` we read the highest nonce, as we aren't assured to actually page that one into memory and start processing it.